go/vt/topotools/routing_rules.go

```diff
@@ -59,6 +69,29 @@ func SaveRoutingRules(ctx context.Context, ts *topo.Server, rules map[string][]s
 	return ts.SaveRoutingRules(ctx, rrs)
 }
 
+//endregion
+
+//region shard routing rules
+
+func GetShardRoutingRuleKey(fromKeyspace, shard string) string {
+	return fmt.Sprintf("%s.%s", fromKeyspace, shard)
+}
+func ParseShardRoutingRuleKey(key string) (string, string) {
+	arr := strings.Split(key, ".")
+	return arr[0], arr[1]
```

Comment:
We should probably confirm that `arr` has the expected number of elements and return an error if not. Otherwise we'll crash with an index out of bounds.